packages/wonder-blocks-cell/src/util/types.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import * as React from "react";
+
+import type {Typography} from "@khanacademy/wonder-blocks-typography";
+
+/**
+ * An element or component that represents an accessory within a cell.
+ */
+type Accessory = React.Node;
+
+/**
+ * A subset of CSS Properties to allow overriding some of the default styles set
+ * on the accessory wrapper (loosely based on StyleType).
+ */
+export type AccessoryStyle = {|
+    /**
+     * A subset of valid Spacing values.
+     */
+    minWidth?: 16 | 24 | 32 | 48,
+
+    /**
+     * To vertically align the accessory.
+     */
+    alignSelf?: "flex-start" | "flex-end" | "center",
+
+    /**
+     * To horizontally align the accessory.
+     */
+    alignItems?: "flex-start" | "flex-end" | "center",
+|};
+
+/**
+ * A union that allows using  plain text or WB Typography elements.
+ */
+export type TypographyText = string | React.Element<Typography>;
+
+/**
+ * Common properties for all cells.
+ */
+export type CellProps = {|
+    /**
+     * The title / main content of the cell. You can either provide a string or
+     * a Typography component.
+     */
+    title: TypographyText,
+
+    /**
```

Comment:
I started with disjoint union types, but things got so overcomplicated, that I feel that it's not worth the effort having to support different cases for optional styles, so I decided to use a simple object with optional props instead.

You can see the original/initial implementation here: https://github.com/Khan/wonder-blocks/pull/1275/commits/f9fd2fa646a7b75179947dc6b8dbd7bd2c07c882#diff-ff5063c238d05911e71992d6aebc1d6f8275e578966cb8858975b6d520e8de54L41

Reply:
Yeah, I think that's reasonable - the disjoint version was very complicated!